### PR TITLE
Port tangle changes to stardust

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by bingyanglin — DISMISSED** [review]

**Review by samuel-rufi — DISMISSED** [review]

File: bee-api/bee-rest-api/src/endpoints/routes/api/v2/transaction_included_message.rs [hunk 1]
```diff
@@ -43,11 +43,13 @@ pub(crate) fn filter<B: StorageBackend>(
         ))
         .and(with_storage(storage))
         .and(with_tangle(tangle))
-        .and_then(transaction_included_message)
+        .and_then(|transaction_id, storage, tangle| async move {
+            transaction_included_message(transaction_id, storage, tangle)
+        })
```
Comment (Alex6323) [inline bee-api/bee-rest-api/src/endpoints/routes/api/v2/transaction_included_message.rs:48]: I think you can remove the `async move` closure for now.

File: bee-api/bee-rest-api/src/endpoints/mod.rs [hunk 2]
```diff
@@ -105,7 +105,7 @@ where
                 requested_messages,
                 consensus_worker,
             )
-            .recover(handle_rejection);
+            .recover(|err| async { handle_rejection(err) });
```
Comment (Alex6323) [inline bee-api/bee-rest-api/src/endpoints/mod.rs:108]: remove `async` block.

File: bee-api/bee-rest-api/src/endpoints/mod.rs [hunk 3]
```diff
@@ -170,7 +170,9 @@ where
         node.spawn::<Self, _, _>(|shutdown| async move {
             info!("Running.");
 
-            let health = warp::path("health").map(|| StatusCode::OK).recover(handle_rejection);
+            let health = warp::path("health")
+                .map(|| StatusCode::OK)
+                .recover(|err| async { handle_rejection(err) });
```
Comment (Alex6323) [inline bee-api/bee-rest-api/src/endpoints/mod.rs:175]: same

File: bee-node/bee-plugin/bee-plugin-dashboard/src/routes.rs [hunk 4]
```diff
@@ -50,25 +50,27 @@ fn serve_asset(path: &str) -> Result<impl Reply, Rejection> {
...
 
 pub(crate) fn asset_routes() -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
     warp::path("branding")
         .and(warp::path::full())
-        .and_then(serve_full_path)
-        .or(warp::path("static").and(warp::path::full()).and_then(serve_full_path))
+        .and_then(|path| async move { serve_full_path(path) })
+        .or(warp::path("static")
+            .and(warp::path::full())
+            .and_then(|path| async move { serve_full_path(path) }))
```
Comment (Alex6323) [inline bee-node/bee-plugin/bee-plugin-dashboard/src/routes.rs:62]: here as well

File: bee-protocol/src/workers/index_updater.rs [hunk 5]
```diff
@@ -116,17 +114,14 @@ async fn update_past_cone<B: StorageBackend>(
...
-                // TODO: That was fine in a synchronous scenario, where this algo had the newest information, but
-                // probably isn't the case in the now asynchronous scenario. Investigate!
-                metadata.set_omrsi(IndexId::new(index, parent_id));
-                metadata.set_ymrsi(IndexId::new(index, parent_id));
-            })
-            .await;
-
-        if let Some(parent) = tangle.get(&parent_id).await {
+        tangle.update_metadata(&parent_id, |metadata| {
+            metadata.set_milestone_index(index);
```
Comment (Alex6323) [inline bee-protocol/src/workers/index_updater.rs:118]: Just as a reminder that we might want to add it here as well: PR #1255 adds a trace log here to allow monitoring this `update_metadata` call.

Comment (thibault-martinez) [reply bee-protocol/src/workers/index_updater.rs:118]: Yeah there is probably way more missing than just trace logs tbh, we'll resolve that when we do the final production/shimmer merge 👍

File: bee-storage/bee-storage-rocksdb/Cargo.toml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "bee-storage-rocksdb"
-version = "0.5.0"
+version = "0.7.0"
```
Comment (Alex6323) [inline bee-storage/bee-storage-rocksdb/Cargo.toml]: Shouldn't this be `0.8.0`? Something is very confusing to me with the version juggling here. Haha

Comment (thibault-martinez) [reply bee-storage/bee-storage-rocksdb/Cargo.toml]: Done

**Review by Alex6323 — APPROVED** [review]

**Review by bingyanglin — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**